src/types.ts

```diff
@@ -6,6 +6,7 @@ export interface ResourceMeta {
   assets?: string[]
   isEntry?: boolean
   isDynamicEntry?: boolean
+  selfRegister?: boolean
```

Comment:
Wdyt to rename to something like `sideEffects: true` (because on any side-effect, order matters).

Reply:
nice idea